### ISSUE TITLE
Add rule to warn on unused definitions

### ISSUE THREAD

**Issue by wooorm** [wooorm/remark-lint#34]
```md
<!-- Good -->
[foo][]

[foo]: https://example.com

<!-- Bad -->
[bar]: https://example.com
```
